### DISCUSSION TITLE
Rearrange prompt.

### DISCUSSION THREAD

**Discussion by SplitMilky** [adieyal/comfyui-dynamicprompts#32] [Q&A]
Is there the ability to rearrange the prompt.  I want to just move the different words or statements around inside the prompt.
For example.  
Cat, car, dog, would iterate through all possible combinations.

cat, dog, car
dog, cat, car
dog, car, cat... and so on.

Also having the ability to keep cat at the front and integrate through the others like so maybe.
cat, {dog, car, ball}
cat, dog, car, ball
cat, dog, ball, car
cat, ball, car, dog
cat, ball, dog, car.. and so on

**Comment by adieyal** [comment]
If you're okay with duplicates, then simply

`{cat|dog|car} {cat|dog|car} {cat|dog|car}` would do the trick

you could also use a [cyclical sampler](https://github.com/adieyal/sd-dynamic-prompts/blob/main/docs/SYNTAX.md#cyclical-sampler), it won't give you all the permutations though, e.g.

`{@cat|dog|car} {@dog|car|cat} {@car|cat|dog}`
this will generate
cat dog car
dog car cat
car cat dog

You might be able to do something with [Jinja templates](https://github.com/adieyal/sd-dynamic-prompts/blob/main/jinja2.md) as well but that might be a little more difficult.
